File: LeetCode/491递增子序列.java
class Solution {
    List<List<Integer>> ans = new ArrayList<>();
    LinkedList<Integer> path = new LinkedList<>();

    public List<List<Integer>> findSubsequences(int[] nums) {
        backtracking(nums, 0);
        return ans;
    }

    public void backtracking(int[] nums, int ind) {
        if (path.size() > 1) ans.add(new ArrayList<>(path));
        HashMap<Integer, Integer> hash = new HashMap<>();
        for (int i = ind; i < nums.length; i++) {
            if (!path.isEmpty() && nums[i] < path.getLast()) continue;
            if (hash.getOrDefault(nums[i], 0) >= 1) continue;
            hash.put(nums[i], hash.getOrDefault(nums[i], 0) + 1);
            path.add(nums[i]);
            backtracking(nums, i + 1);
            path.removeLast();
        }
    }
}

File: LeetCode/491递增子序列.go
var ans [][]int
var path []int

func findSubsequences(nums []int) [][]int {
	ans = [][]int{}
	path = []int{}
	backtracking(nums, 0)
	return ans
}

func backtracking(nums []int, ind int) {
	if len(path) > 1 {
		temp := make([]int, len(path))
		copy(temp, path)
		ans = append(ans, temp)
	}
	cnt := make([]int, 205)
	for i := ind; i < len(nums); i++ {
		if len(path) > 0 && nums[i] < path[len(path)-1] || cnt[nums[i]+100] == 1 {
			continue
		}
		cnt[nums[i]+100] = 1
		path = append(path, nums[i])
		backtracking(nums, i+1)
		path = path[:len(path)-1]
	}
}